packages/components/src/confirm-dialog/types.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { SyntheticEvent } from 'react';
+
+interface Props {
+	isOpen?: undefined;
```

Comment:
`isOpen` might be `undefined` though. I explained it [here](https://github.com/WordPress/gutenberg/pull/34153#discussion_r715701219).

Reply:
> isOpen might be undefined though

Yes, but typing it as `isOpen?: undefined;` means that it can _only_ be undefined — either by being omitted (`?` notation) or by being explicitly assigned to `undefined`

Reply:
Ah yeah, I see what you mean. It's indeed weird as it's an optional `undefined` type, which indeed doesn't make much sense. I'll remove the line and TS should still detect the proper type structure in the union.